### PR TITLE
Update Go to 1.21 and upgrade deps

### PR DESCRIPTION
**Pull request by tucksaun**
Blocked by https://github.com/symfony-cli/cert/pull/5, https://github.com/symfony-cli/console/pull/5, https://github.com/symfony-cli/dumper/pull/3 and https://github.com/symfony-cli/phpstore/pull/14

### CODE REVIEW

**Comment by fabpot** [comment]
Unblocked

File: go.mod [hunk 1]
```diff
@@ -21,26 +21,28 @@ require (
 	github.com/stoicperlman/fls v0.0.0-20171222144224-f073b7a01081
 	github.com/symfony-cli/cert v1.0.2
 	github.com/symfony-cli/console v1.0.3
+	github.com/symfony-cli/dumper v1.0.1
```
Comment (fabpot) [inline go.mod:24]: I think this is a mistake, right?

File: go.sum [hunk 2]
```diff
@@ -5,23 +5,33 @@ github.com/NYTimes/gziphandler v1.1.1 h1:ZUDjpQae29j0ryrS0u/B8HZfJBtBQHjqw2rQ2cq
 github.com/NYTimes/gziphandler v1.1.1/go.mod h1:n/CVRwUEOgIxrgPvAQhUUr9oeUtvrhMomdKFjzJNB0c=
 github.com/compose-spec/compose-go v1.17.0 h1:cvje90CU94dQyTnJoHJYjx9yE4Iggse1XmGcO3Qi5ts=
 github.com/compose-spec/compose-go v1.17.0/go.mod h1:zR2tP1+kZHi5vJz7PjpW6oMoDji/Js3GHjP+hfjf70Q=
```
Comment (fabpot) [inline go.sum:7]: Maybe we can run `mod tidy`?

File: go.mod (hunk 1 above)
Comment (tucksaun) [reply go.mod]: indeed!
fixed

File: go.sum (hunk 2 above)
Comment (tucksaun) [reply go.sum]: 👍 done

**Comment by fabpot** [comment]
Thank you @tucksaun

**Pull request merged**